src/storage/src/sink/healthcheck.rs

```diff
@@ -111,64 +28,62 @@ pub enum SinkStatus {
         /// the `mz_sink_status_history` table.
         hint: Option<String>,
     },
-    /// Represents a irrecoverable failure in the pipeline. Data from this collection
-    /// is not queryable any longer. The only valid transition from Failed is Dropped.
-    Failed {
-        /// Error string used to populate the `error` column in the `mz_sink_status_history` table.
-        error: String,
-        /// Optional hint string which if present, will be added to the `details` column in
-        /// the `mz_sink_status_history` table.
-        hint: Option<String>,
-    },
-    /// Represents a sink that was dropped.
-    Dropped,
+    // Managed by the storage controller.
+    // Dropped,
+}
+
+impl crate::healthcheck::HealthStatus for SinkStatus {
+    fn name(&self) -> &'static str {
+        self.name()
+    }
+    fn error(&self) -> Option<&str> {
+        self.error()
+    }
+    fn hint(&self) -> Option<&str> {
+        self.hint()
+    }
+    fn should_halt(&self) -> bool {
+        matches!(self, Self::Stalled { .. })
+    }
+    fn can_transition_from(&self, other: Option<&Self>) -> bool {
+        Self::can_transition(other, self)
+    }
+    fn starting() -> Self {
+        Self::Starting
+    }
 }
 
 impl SinkStatus {
     fn name(&self) -> &'static str {
         match self {
-            SinkStatus::Setup => "setup",
             SinkStatus::Starting => "starting",
             SinkStatus::Running => "running",
             SinkStatus::Stalled { .. } => "stalled",
-            SinkStatus::Failed { .. } => "failed",
```

Comment:
I am assuming this is removed because it's not actually used.

Reply:
yep!